src/routes/charts.js

```
@@ -926,7 +945,9 @@ async function writeChartAsset(request, h) {
     const chart = await loadChart(request);
 
     const isGuestChart = chart.guest_session === request.auth.credentials.session;
-    const isEditable = isGuestChart || (await chart.isEditableBy(request.auth.artifacts));
+    const isEditable =
+        isGuestChart ||
```

same here, no need for `isGuestChart ||` since `Chart.isEditableBy` is handling guest sessions as well.

---

though `chart.isEditableBy` can only do that if the session is passed in as the second parameter. this was a bug that is frequent across our codebase. maybe the function should throw an exception if called with only a user?

---

I don't think this would make sense, as the function can also being called in scenarios without session (eg an API request authenticated using token). It should be easy to identify all places where this function is being used and ensure that the session parameter is passed.